### PR TITLE
refactor: unify episode generation under one orchestrator

### LINKED ISSUE #31: SP-001: Unify episode generation into single orchestrator
## User intent
Consolidate duplicate/dead-end logic in the question gen -> episode gen path while preserving `template`, `procedural`, and `llm_gen`, with fail-fast behavior.

## Problem statement
- Episode generation is split across `src/datagen/episode_gen.py` and `src/datagen/validate_synthetic.py` with duplicated task prep, manifest filtering, session handling, and output writing.
- Runtime schema usage is inconsistent (`question_text` vs legacy/mechanical fallbacks), creating execution risk and contract drift.

## Approved approach
- Keep `src/datagen/episode_gen.py` as the single episode orchestrator for all sources.
- Remove `src/datagen/validate_synthetic.py`.
- Implement strategy-based verification in the unified orchestrator:
  - `llm_gen` -> consistency triangulation.
  - `template|procedural` -> single-trace ground-truth verification.
- Standardize runtime question prompt resolution to:
  - first `question_text`
  - then `question_mechanical`
  - never legacy `question` key in runtime orchestration paths.

## File touch scope
- `src/datagen/episode_gen.py`
- `src/datagen/validate_synthetic.py` (delete)
- `src/datagen/pipeline.py`
- `src/cli.py`
- `src/datagen/shared/verification.py`
- `src/datagen/shared/episode_factory.py`
- `src/datagen/teacher.py`
- tests directly validating these contract surfaces

## Non-goals
- Training loop/model training changes.
- Changing source taxonomy (`template`, `procedural`, `llm_gen`).
- Introducing backward-compat shims for deprecated schema keys.

## Acceptance + regression
- All episode generation entrypoints run through one module path.
- `template`, `procedural`, and `llm_gen` all produce episodes through the unified orchestrator.
- No runtime dependence on legacy `question` key in the unified episode path.
- Lint and tests pass for touched surfaces.

## Risks + rollback trigger
- Risk: mixed-source regressions due to orchestration merge.
- Rollback trigger: source-specific episode counts/verification rates regress materially vs baseline on small validation batch.
- Rollback action: revert to prior split only if blocking regression cannot be fixed quickly in-branch.

## Overlap decision
- This issue owns episode-path unification and runtime schema-key normalization.
- GUI removal and stale tooling deletion are handled in a separate issue.

## Manual approval evidence summary
- 2026-02-17 user approved direction and constraints, including bold refactor.
- 2026-02-17 user approved the unification proposal and requested issue creation.

### PR DESCRIPTION
**Pull request by thajpo**
## Summary
- route `template`, `procedural`, and `llm_gen` episode generation through one unified `src/datagen/episode_gen.py` entrypoint with source-specific verification strategy
- remove `src/datagen/validate_synthetic.py` and update `src/cli.py` + `src/datagen/pipeline.py` to use the unified module path
- standardize runtime prompt resolution (`question_text` -> `question_mechanical`) across episode verification/factory/teacher paths and remove legacy episode-factory wrappers
- update regression tests to enforce the new single-entrypoint contract and removed module

## Testing
- `uv run ruff check src/datagen/episode_gen.py src/datagen/pipeline.py src/datagen/shared/verification.py src/datagen/shared/episode_factory.py src/datagen/teacher.py src/cli.py tests/test_cli_entrypoint_contract.py tests/test_episode_factory.py tests/test_residual_contract_purge.py`
- `uv run pytest tests/test_cli_entrypoint_contract.py tests/test_episode_factory.py tests/test_residual_contract_purge.py tests/test_artifact_path_resolver.py tests/test_pipeline_e2e.py -rs`
- `uv run pytest tests/test_strict_answer_contract.py tests/test_unified_schema.py tests/test_triangulation.py`
- `uv run python -m src.datagen.episode_gen --help`

Closes #31

### CODE REVIEW

**Comment by thajpo** [comment]
## Agent Update
- scope addressed: unify episode generation into one orchestrator for `template`, `procedural`, and `llm_gen` (issue #31)
- changes made: removed `validate_synthetic`, merged synthetic/llm flows into `episode_gen` strategy routing, standardized prompt resolution (`question_text` -> `question_mechanical`), updated CLI/pipeline routing, and refreshed contract tests
- files touched: `src/datagen/episode_gen.py`, `src/datagen/pipeline.py`, `src/cli.py`, `src/datagen/shared/verification.py`, `src/datagen/shared/episode_factory.py`, `src/datagen/teacher.py`, `tests/test_cli_entrypoint_contract.py`, `tests/test_episode_factory.py`, `tests/test_residual_contract_purge.py`, deleted `src/datagen/validate_synthetic.py`
- commands run: ruff check on touched files; focused pytest suites (`test_cli_entrypoint_contract`, `test_episode_factory`, `test_residual_contract_purge`, `test_artifact_path_resolver`, `test_pipeline_e2e`); verification/triangulation suites (`test_strict_answer_contract`, `test_unified_schema`, `test_triangulation`); module CLI smoke (`python -m src.datagen.episode_gen --help`)
- test results: all listed suites passed (75 passed total, 1 pre-existing skipped test with explicit marker)
- remaining open items: none for #31 in-scope contract; GUI references to removed module are intentionally deferred to #32
- risks/assumptions: assumes GUI path changes are handled separately in #32; this PR intentionally does not touch GUI files to stay issue-scoped
